tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,263 @@
+PORT = 9000
+UUID = "test-uuid"
+UNKNOWN_ERROR = "1234"
+
+
+async def mock_discover(_discovery_callback):
+    """Mock discovering a Logitech Media Server."""
+    _discovery_callback(Server(None, HOST, PORT, uuid=UUID))
+
+
+async def mock_failed_discover(_discovery_callback):
+    """Mock unsuccessful discovery by doing nothing."""
+
+
+async def patch_async_query_unauthorized(self, *args):
+    """Mock an unauthorized query."""
+    self.http_status = HTTP_UNAUTHORIZED
+    return False
+
+
+async def test_user_form(hass):
+    """Test user-initiated flow, including discovery and the edit step."""
+    with patch("pysqueezebox.Server.async_query", return_value={"uuid": UUID},), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.squeezebox.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover", mock_discover
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+        assert CONF_HOST in result["data_schema"].schema
+        for key in result["data_schema"].schema:
+            if key == CONF_HOST:
+                assert key.description == {"suggested_value": HOST}
+
+        # test the edit step
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: HOST, CONF_PORT: PORT, CONF_USERNAME: "", CONF_PASSWORD: ""},
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == HOST
+        assert result["data"] == {
+            CONF_HOST: HOST,
+            CONF_PORT: PORT,
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+        }
+
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_form_timeout(hass):
+    """Test we handle server search timeout."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover",
+        mock_failed_discover,
+    ), patch("homeassistant.components.squeezebox.config_flow.TIMEOUT", 0.1):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "no_server_found"}
+
+        # simulate manual input of host
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_HOST: HOST2}
+        )
+        assert result2["type"] == RESULT_TYPE_FORM
+        assert result2["step_id"] == "edit"
+        assert CONF_HOST in result2["data_schema"].schema
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)